ADA/ada.qmd

```diff
@@ -19,57 +17,234 @@ It brings together:
+
+For regular analysis pipelines adopting RAP principles we'd usually advise against using notebooks and instead recommend a Git controlled repository of code scripts using the [standard repository template available in dfeR](https://dfe-analytical-services.github.io/dfeR/reference/create_project.html). This is because version control can be more difficult in Databricks notebooks, as there is the potential for multiple people to make edits to the same notebook at once. Additionally, with a full script in an IDE, when the code is run without modifications it will provide the same output consistently. However, with notebooks you can run code cells independently of one another. If you don't run all code cells in order then your output could be inconsistent and result in errors, potentially making QA more difficult. 
+
+
+:::{.callout-warning}
+When using Databricks notebooks with git, cell outputs are automatically cleared by default when your code is pushed up to a repository. However, if you share a notebook using the Databricks interface, cell outputs remain intact from the last time you ran the code. This risks sharing data that should not be shared with others. Before sharing code with someone outside your team, you must use the `Clear all cell outputs` option in the Databricks "Run" menu to avoid inadvertently sharing data that only you and your team should have access to. 
+:::
+
+If the rest of your team are unfamiliar with notebooks and will need to pick up your work at a later date, this could cause issues. The [Statistics Development Team](mailto:statistics.development@education.gov.uk) are happy to advise on this and provide guidance if you have any concerns. 
+
+
+You can read more about the differences between IDEs and notebooks and where each might be useful [on the NHS-R website](/ADA/ada.html#differences-between-ides-and-notebooks).
+
+---
+
+#### Writing and running code: scripts and IDEs
+
+---
+
+Scripts (e.g. R scripts) can be created and saved in your Workspace and run via the Databricks interface. 
+
+There is detailed guidance on using the Databricks SQL editor, which is similar to SSMS, on the [ADA guidance site](https://analytical-data-access.education.gov.uk/get-support/using-sql-in-databricks-sql-editor).
+
+One benefit of Databricks is the use of workflows to run your code. Workflows allow you to automate the running of single or multiple scripts, so if you have large amounts of data processing to do on a regular basis, you can set up a workflow to complete this overnight. Your laptop does not need to be switched on for this to work. 
```

Comment:
Should workflows get their own header and section? If so, I'd take just this paragraph tweak it a little and stick it under a new header: 'Writing and running code: workflows'

Something like this maybe?
> Databricks allows the use of workflows to run your code. 
>
> Workflows allow you to schedule and automate the running of code scripts, so if you have large amounts of data processing to do on a regular basis, you can set up a workflow to complete this overnight. Your laptop does not need to be switched on for this to work.
>
> Link to more detailed workflows guidance if they want to learn more?

Reply:
Yeah could do - the workflows guidance isn't finished yet so I also need to get that done